pkg/storage/enhanced/azblob.go

```diff
@@ -0,0 +1,319 @@
+package enhanced
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/Altinity/clickhouse-backup/v2/pkg/config"
+	"github.com/Altinity/clickhouse-backup/v2/pkg/storage"
+	"github.com/Azure/azure-storage-blob-go/azblob"
+	"github.com/rs/zerolog/log"
+)
+
+// EnhancedAzureBlob implements BatchRemoteStorage for Azure Blob Storage with parallel operations
+type EnhancedAzureBlob struct {
+	storage.RemoteStorage
+	container azblob.ContainerURL
+	config    *config.AzureBlobConfig
+	metrics   *DeleteMetrics
+}
+
+// AzureBlobDeleteJob represents a delete job for parallel processing
+type AzureBlobDeleteJob struct {
+	blobName string
+	result   chan AzureBlobDeleteResult
+}
+
+// AzureBlobDeleteResult represents the result of a delete operation
+type AzureBlobDeleteResult struct {
+	blobName string
+	error    error
+}
+
+// AzureBlobWorkerPool manages parallel delete workers
+type AzureBlobWorkerPool struct {
```

Comment:
why we trying generate separate pool implementation instead of pool libs use exists in project or using errgroup.WithContext with exists rertier?